Modules: new module "park"

AndKe opened the pull request.

stephendade commented:
We've recently pushed a fix to the MAVProxy CI. You'll need to rebase to get the CI working for this PR.

AndKe commented:
@stephendade  - thank you, I had no clue that adding one file, could somehow depend on that other file.
I've tried to update, but I see no "results" in this PR.
Can you please point out my error?

```
andre@loke:~/prog/MAVProxy$ git checkout master 
Switched to branch 'master'
andre@loke:~/prog/MAVProxy$ **git pull upstream HEAD** 
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 8 (delta 4), reused 5 (delta 3), pack-reused 0
Unpacking objects: 100% (8/8), 917 bytes | 131.00 KiB/s, done.
From github.com:Dronecode/MAVProxy
 * branch              HEAD       -> FETCH_HEAD
Updating 8fa66b40..435d407d
Fast-forward
 **.github/workflows/windows_build.yml** | 4 ++++
 README.md                           | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)
andre@loke:~/prog/MAVProxy$ git submodule update 
andre@loke:~/prog/MAVProxy$ git checkout module-park 
Switched to branch 'module-park'
andre@loke:~/prog/MAVProxy$ **git rebase upstream/master** 
Current branch module-park is up to date.

andre@loke:~/prog/MAVProxy$ git push origin module-park 
**Everything up-to-date**
andre@loke:~/prog/MAVProxy$ 

$ **git log** 
commit 90bc9b19aa5c4b7d6f6ced60c61e43d8687b1c54 (HEAD -> module-park, origin/module-park)
Author: Andre Kjellstrup <andre.kjellstrup@gmail.com>
Date:   Thu Feb 10 14:59:33 2022 +0100

    park module

commit 8fa66b402e96f4d0a2503e0e31e4427bbadf4359 (upstream/master)
Author: Andrew Tridgell <andrew@tridgell.net>
Date:   Mon Feb 7 13:12:56 2022 +1100

    make CAN_FRAME a bit nicer to use


```

stephendade commented:
>  Can you please point out my error?

Huh ... not too sure why it's not working. I usually do ``git pull --rebase upstream master`` to grab new changes into a branch.

AndKe commented:
@stephendade I tried your command, and now got yesterday's commit too.
As you can see from the git log output in both cases - mine was not "outdated" before.
```
$ git pull --rebase upstream master
From github.com:Dronecode/MAVProxy
 * branch              master     -> FETCH_HEAD
   8fa66b40..435d407d  master     -> upstream/master
Successfully rebased and updated refs/heads/module-park.
andre@loke:~/prog/MAVProxy$ git log -2
commit 2e3f85695a5c7de11e150a6230e1b7e031930c96 (HEAD -> module-park)
Author: Andre Kjellstrup <andre.kjellstrup@gmail.com>
Date:   Thu Feb 10 14:59:33 2022 +0100

    park module

commit 435d407d844d455ea7f6bb1813d5860a163abce4 (upstream/master, origin/master, origin/HEAD, master)
Author: Stephen Dade <stephen_dade@hotmail.com>
Date:   Fri Feb 11 11:33:37 2022 +1100

```

tridge commented on MAVProxy/modules/mavproxy_park.py:
```diff
@@ -0,0 +1,113 @@
+            self.lon = m.lon * 1.0e-7
+            self.alt = m.relative_alt * 1.0e-3
+
+    def haversine(self, lon1, lat1, alt1, lon2, lat2, alt2):
```
you can just call mp_util.gps_distance() to get the horizontal component

AndKe replied:
@tridge now I remember:  I did not use mp_util.gps_distance() because it only calculated lateral distance, while I wished to have a 3D sphere-shaped shape - for correct warning in all directions.
Of course, I could check for altitude separately, ending with a cylindrical threshold (tripping on a lateral distance or vertical deviation.) 
Are there drawbacks of doing it the way I did?

tridge replied:
it looks like you calculate horizontal then add vertical, so you can use gps_distance for horizontal and add vertical in the same way
I just like all our earth model fns to be consistent, I'm not keen on having two different methods in the code for this distance.

AndKe replied:
Trank you for replying, I'll change it.

AndKe replied:
@tridge   - Please accept the fixed version.
